python_modules/dagster/dagster/core/definitions/event_metadata/__init__.py

```diff
@@ -95,6 +96,20 @@ def parse_metadata(
             metadata_entries, "metadata_entries", (EventMetadataEntry, PartitionMetadataEntry)
         )
 
+    # This is a stopgap measure to deal with unsupported metadata values, which occur when we try
+    # to convert arbitrary metadata (on e.g. OutputDefinition) to EventMetadata, which is required
+    # for serialization. This will cause unsupported values to be silently dropped-- eventually we
+    # should probably standardize the metadata system across dagster.
+    elif drop_invalid:
+        metadata_entries = []
+        for k, v in metadata.items():
+            try:
+                metadata_entries.append(parse_metadata_entry(k, v))
+            except DagsterInvalidEventMetadata:
+                pass
+        return metadata_entries
```

Comment:
Is this under test? Whats an example of an object that gets dropped? 

should we leave some entry at the key where the value communicates it wasn't not serializable?

Reply:
>Is this under test?

Not right now.

>Whats an example of an object that gets dropped?

Anything that there is not an existing `EventMetadataEntry` variant for. For example, Python lists (even if they just contain primitives) are not currently supported. So `columns` in this snippet from the hacker_news repo gets dropped:

```
@op(
    ins={
        "stories": In(
            root_manager_key="warehouse_loader",
            metadata={"table": "hackernews.stories", "columns": ["id"]},
        ),
        "comments": In(
            root_manager_key="warehouse_loader",
            metadata={
                "table": "hackernews.comments",
                "columns": ["id", "user_id", "parent"],
            },
        ),
    },
    out=Out(
        io_manager_key="warehouse_io_manager", metadata={"table": "hackernews.comment_stories"}
    ),
)

>should we leave some entry at the key where the value communicates it was not serializable?

That sounds like a good idea, I'll add it.
```

Reply:
>  do you see any issues with the handling of those values? I added testing and coercion to placeholder text

What you added looks good - just need to rename `drop_invalid`

Reply:
>just need to rename drop_invalid

What would you suggest? `coerce_invalid`? I kept `drop` because even though we are keeping the keys, the _values_ are being dropped (the placeholder values have little semantic relation to the originals).

Reply:
ah just seemed vestigial

`drop` keeping keys feels odd to me personally. Spitballing: `coerce_invalid` , `coerce_unserializable`, `allow_unserializable`, `allow_invalid`, `allow_unhandled`, `raise_on_unknown` (with bool flipped)

Reply:
@alangenfeld good to merge?